File: javascript/src/model/PlateDetails.js
// require files in Node.js environment
var Coordinate;var PlateCandidate;
if (typeof module === 'object' && module.exports) {
  
  Coordinate = require('./Coordinate.js');
  PlateCandidate = require('./PlateCandidate.js');
}




//export module
if ( typeof define === "function" && define.amd ) {
  define('PlateDetails', ['jquery', 'Array', 'Array'],
    function($, ArrayArray) {
      return PlateDetails;
   });
}


var PlateDetails = function PlateDetails() { 
  var self = this;
  
  /**
   * Best plate number for this plate
   * datatype: String
   **/
  self.plate = null;
  
  /**
   * Indicates whether the plate matched a regional text pattern
   * datatype: Integer
   **/
  self.matchesTemplate = null;
  
  /**
   * The max number of results requested
   * datatype: Integer
   **/
  self.requestedTopn = null;
  
  /**
   * Number of milliseconds to process the license plate
   * datatype: Number
   **/
  self.processingTimeMs = null;
  
  /**
   * Confidence percentage that the plate number is correct
   * datatype: Number
   **/
  self.confidence = null;
  
  /**
   * Specified or detected region (e.g., tx for Texas)
   * datatype: String
   **/
  self.region = null;
  
  /**
   * Confidence percentage that the plate region is correct
   * datatype: Number
   **/
  self.regionConfidence = null;
  
  /**
   * The X/Y coordinates of the license plate in the image\nFour coordinates are provided that form a polygon starting\nfrom the top-left and moving clockwise ending in the bottom left
   * datatype: Array
   **/
  self.coordinates = [];
  
  /**
   * All the top N candidates that could be the correct plate number
   * datatype: Array
   **/
  self.candidates = [];
  
  
  self.constructFromObject = function(data) {
    if (!data) {
      return;
    }
    
    self.plate = data.plate;
    
    self.matchesTemplate = data.matches_template;
    
    self.requestedTopn = data.requested_topn;
    
    self.processingTimeMs = data.processing_time_ms;
    
    self.confidence = data.confidence;
    
    self.region = data.region;
    
    self.regionConfidence = data.region_confidence;
    
    self.coordinates = new Array();
    
    self.candidates = new Array();
    
  }

  
  /**
   * get Best plate number for this plate
   * @return {String}
   **/
  self.getPlate = function() {
    return self.plate;
  }

  /**
   * set Best plate number for this plate
   * @param {String} plate
   **/
  self.setPlate = function (plate) {
    self.plate = plate;
  }
  
  /**
   * get Indicates whether the plate matched a regional text pattern
   * @return {Integer}
   **/
  self.getMatchesTemplate = function() {
    return self.matchesTemplate;
  }

  /**
   * set Indicates whether the plate matched a regional text pattern
   * @param {Integer} matchesTemplate
   **/
  self.setMatchesTemplate = function (matchesTemplate) {
    self.matchesTemplate = matchesTemplate;
  }
  
  /**
   * get The max number of results requested
   * @return {Integer}
   **/
  self.getRequestedTopn = function() {
    return self.requestedTopn;
  }

  /**
   * set The max number of results requested
   * @param {Integer} requestedTopn
   **/
  self.setRequestedTopn = function (requestedTopn) {
    self.requestedTopn = requestedTopn;
  }
  
  /**
   * get Number of milliseconds to process the license plate
   * @return {Number}
   **/
  self.getProcessingTimeMs = function() {
    return self.processingTimeMs;
  }

  /**
   * set Number of milliseconds to process the license plate
   * @param {Number} processingTimeMs
   **/
  self.setProcessingTimeMs = function (processingTimeMs) {
    self.processingTimeMs = processingTimeMs;
  }
  
  /**
   * get Confidence percentage that the plate number is correct
   * @return {Number}
   **/
  self.getConfidence = function() {
    return self.confidence;
  }

  /**
   * set Confidence percentage that the plate number is correct
   * @param {Number} confidence
   **/
  self.setConfidence = function (confidence) {
    self.confidence = confidence;
  }
  
  /**
   * get Specified or detected region (e.g., tx for Texas)
   * @return {String}
   **/
  self.getRegion = function() {
    return self.region;
  }

  /**
   * set Specified or detected region (e.g., tx for Texas)
   * @param {String} region
   **/
  self.setRegion = function (region) {
    self.region = region;
  }
  
  /**
   * get Confidence percentage that the plate region is correct
   * @return {Number}
   **/
  self.getRegionConfidence = function() {
    return self.regionConfidence;
  }

  /**
   * set Confidence percentage that the plate region is correct
   * @param {Number} regionConfidence
   **/
  self.setRegionConfidence = function (regionConfidence) {
    self.regionConfidence = regionConfidence;
  }
  
  /**
   * get The X/Y coordinates of the license plate in the image\nFour coordinates are provided that form a polygon starting\nfrom the top-left and moving clockwise ending in the bottom left
   * @return {Array}
   **/
  self.getCoordinates = function() {
    return self.coordinates;
  }

  /**
   * set The X/Y coordinates of the license plate in the image\nFour coordinates are provided that form a polygon starting\nfrom the top-left and moving clockwise ending in the bottom left
   * @param {Array} coordinates
   **/
  self.setCoordinates = function (coordinates) {
    self.coordinates = coordinates;
  }
  
  /**
   * get All the top N candidates that could be the correct plate number
   * @return {Array}
   **/
  self.getCandidates = function() {
    return self.candidates;
  }

  /**
   * set All the top N candidates that could be the correct plate number
   * @param {Array} candidates
   **/
  self.setCandidates = function (candidates) {
    self.candidates = candidates;
  }
  

  self.toJson = function () {
    return JSON.stringify(self);
  }
}

if (typeof module === 'object' && module.exports) {
  module.exports = PlateDetails;
}

File: java/src/main/java/com/openalpr/api/models/PlateDetails.java
package com.openalpr.api.models;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import com.openalpr.api.models.Coordinate;
import com.openalpr.api.models.PlateCandidate;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.math.BigDecimal;
import java.util.*;





@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-02-17T14:18:32.383-05:00")
public class PlateDetails   {
  
  private String plate = null;
  private Integer matchesTemplate = null;
  private Integer requestedTopn = null;
  private BigDecimal processingTimeMs = null;
  private BigDecimal confidence = null;
  private String region = null;
  private BigDecimal regionConfidence = null;
  private List<Coordinate> coordinates = new ArrayList<Coordinate>();
  private List<PlateCandidate> candidates = new ArrayList<PlateCandidate>();

  
  /**
   * Best plate number for this plate
   **/
  
  @ApiModelProperty(value = "Best plate number for this plate")
  @JsonProperty("plate")
  public String getPlate() {
    return plate;
  }
  public void setPlate(String plate) {
    this.plate = plate;
  }

  
  /**
   * Indicates whether the plate matched a regional text pattern
   **/
  
  @ApiModelProperty(value = "Indicates whether the plate matched a regional text pattern")
  @JsonProperty("matches_template")
  public Integer getMatchesTemplate() {
    return matchesTemplate;
  }
  public void setMatchesTemplate(Integer matchesTemplate) {
    this.matchesTemplate = matchesTemplate;
  }

  
  /**
   * The max number of results requested
   **/
  
  @ApiModelProperty(value = "The max number of results requested")
  @JsonProperty("requested_topn")
  public Integer getRequestedTopn() {
    return requestedTopn;
  }
  public void setRequestedTopn(Integer requestedTopn) {
    this.requestedTopn = requestedTopn;
  }

  
  /**
   * Number of milliseconds to process the license plate
   **/
  
  @ApiModelProperty(value = "Number of milliseconds to process the license plate")
  @JsonProperty("processing_time_ms")
  public BigDecimal getProcessingTimeMs() {
    return processingTimeMs;
  }
  public void setProcessingTimeMs(BigDecimal processingTimeMs) {
    this.processingTimeMs = processingTimeMs;
  }

  
  /**
   * Confidence percentage that the plate number is correct
   **/
  
  @ApiModelProperty(value = "Confidence percentage that the plate number is correct")
  @JsonProperty("confidence")
  public BigDecimal getConfidence() {
    return confidence;
  }
  public void setConfidence(BigDecimal confidence) {
    this.confidence = confidence;
  }

  
  /**
   * Specified or detected region (e.g., tx for Texas)
   **/
  
  @ApiModelProperty(value = "Specified or detected region (e.g., tx for Texas)")
  @JsonProperty("region")
  public String getRegion() {
    return region;
  }
  public void setRegion(String region) {
    this.region = region;
  }

  
  /**
   * Confidence percentage that the plate region is correct
   **/
  
  @ApiModelProperty(value = "Confidence percentage that the plate region is correct")
  @JsonProperty("region_confidence")
  public BigDecimal getRegionConfidence() {
    return regionConfidence;
  }
  public void setRegionConfidence(BigDecimal regionConfidence) {
    this.regionConfidence = regionConfidence;
  }

  
  /**
   * The X/Y coordinates of the license plate in the image\nFour coordinates are provided that form a polygon starting\nfrom the top-left and moving clockwise ending in the bottom left
   **/
  
  @ApiModelProperty(value = "The X/Y coordinates of the license plate in the image\nFour coordinates are provided that form a polygon starting\nfrom the top-left and moving clockwise ending in the bottom left")
  @JsonProperty("coordinates")
  public List<Coordinate> getCoordinates() {
    return coordinates;
  }
  public void setCoordinates(List<Coordinate> coordinates) {
    this.coordinates = coordinates;
  }

  
  /**
   * All the top N candidates that could be the correct plate number
   **/
  
  @ApiModelProperty(value = "All the top N candidates that could be the correct plate number")
  @JsonProperty("candidates")
  public List<PlateCandidate> getCandidates() {
    return candidates;
  }
  public void setCandidates(List<PlateCandidate> candidates) {
    this.candidates = candidates;
  }

  

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PlateDetails plateDetails = (PlateDetails) o;
    return Objects.equals(plate, plateDetails.plate) &&
        Objects.equals(matchesTemplate, plateDetails.matchesTemplate) &&
        Objects.equals(requestedTopn, plateDetails.requestedTopn) &&
        Objects.equals(processingTimeMs, plateDetails.processingTimeMs) &&
        Objects.equals(confidence, plateDetails.confidence) &&
        Objects.equals(region, plateDetails.region) &&
        Objects.equals(regionConfidence, plateDetails.regionConfidence) &&
        Objects.equals(coordinates, plateDetails.coordinates) &&
        Objects.equals(candidates, plateDetails.candidates);
  }

  @Override
  public int hashCode() {
    return Objects.hash(plate, matchesTemplate, requestedTopn, processingTimeMs, confidence, region, regionConfidence, coordinates, candidates);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class PlateDetails {\n");
    
    sb.append("    plate: ").append(toIndentedString(plate)).append("\n");
    sb.append("    matchesTemplate: ").append(toIndentedString(matchesTemplate)).append("\n");
    sb.append("    requestedTopn: ").append(toIndentedString(requestedTopn)).append("\n");
    sb.append("    processingTimeMs: ").append(toIndentedString(processingTimeMs)).append("\n");
    sb.append("    confidence: ").append(toIndentedString(confidence)).append("\n");
    sb.append("    region: ").append(toIndentedString(region)).append("\n");
    sb.append("    regionConfidence: ").append(toIndentedString(regionConfidence)).append("\n");
    sb.append("    coordinates: ").append(toIndentedString(coordinates)).append("\n");
    sb.append("    candidates: ").append(toIndentedString(candidates)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

